File: backend/src/main/java/com/backend/api/resume/dto/response/ResumeReadResponse.java
package com.backend.api.resume.dto.response;

import com.backend.domain.resume.entity.Resume;

public record ResumeReadResponse(
        Long id,
        String skill,
        String activity,
        String career,
        String certification,
        String content,
        String portfolioUrl,
        Long userId
) {
    public static ResumeReadResponse from(Resume resume) {
        return new ResumeReadResponse(
                resume.getId(),
                resume.getSkill(),
                resume.getActivity(),
                resume.getCareer(),
                resume.getCertification(),
                resume.getContent(),
                resume.getPortfolioUrl(),
                resume.getUser().getId()
        );
    }
}

File: backend/src/main/java/com/backend/api/resume/dto/response/ResumeReadResponse.kt
package com.backend.api.resume.dto.response

import com.backend.domain.resume.entity.Resume


data class ResumeReadResponse(
    val id: Long,
    val skill: String?,
    val activity: String?,
    val career: String?,
    val certification: String?,
    val content: String?,
    val portfolioUrl: String?,
    val userId: Long
) {
    companion object {
        fun from(resume: Resume): ResumeReadResponse {
            return ResumeReadResponse(
                resume.id,
                resume.skill,
                resume.activity,
                resume.career,
                resume.certification,
                resume.content,
                resume.portfolioUrl,
                resume.user.id
            )
        }
    }
}
